code/clue.js

```diff
@@ -1,59 +1,237 @@
+  occupation: 'A self-taught musician.',
+  favouriteFood: 'Nothing which had a name'
+};
+
+const Hamburger = {
+  name: 'Hamburger',
+  taste: 'Taste: 9',
+  image: 'assets/hamburger.jpg'
 }
 
-// CREATE OBJECTS FOR ALL THE WEAPONS, ADD MORE CHARACTERISTICS TO THE WEAPONS IF YOU LIKE.
+const RachelTrifle = {
+  name: "Rachel's Trifle",
+  taste: 'Taste: 2',
+  image: 'assets/trifle.jpg'
+}
 
-const rope = {
-  name: 'rope',
-  weight: 10
+const Turkey = {
+  name: 'Turkey',
+  taste: 'Taste: 10',
+  image: 'assets/turkey.jpeg'
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const RedBagel = {
+  name: 'Red Bagel',
+  taste: 'Taste: 6',
+  image: 'assets/bagel.jpg'
+}
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const Pizza = {
+  name: 'Pizza',
+  taste: 'Taste: 7',
+  image: 'assets/pizza.png'
+}
 
-const suspects = [
-  mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+const Muffin = {
+  name: 'Muffin',
+  taste: 'Taste: 4',
+  image: 'assets/muffin.jpg'
+}
+
+const KiwiPie = {
+  name: 'Kiwi Pie',
+  taste: 'Taste: 3',
+  image: 'assets/pie.jpg'
+}
+
+const Sandwich = {
+  name: 'Sandwich',
+  taste:'Taste: 5',
+  image: 'assets/sandwich.jpg'
+}
 
-const weapons = []
+const MonicaDiningRoom = {
+  name: "Monica's Dining Room",
+  image: 'assets/mdr.jpg'
+}
+
+const RossApartment = {
+  name: "Ross' Apartment",
+  image: 'assets/ra.jpg'
+}
 
-const rooms = []
+const ChandlerDiningRoom = {
+  name: "Chandler's Dining Room",
+  image: 'assets/cdr.jpg'
+}
+
+const RachelRoom = {
+  name: "Rachel's Room",
+  image: 'assets/rbr.jpg'
+}
+
+const MonicaRoom = {
+  name: "Monica's Room",
+  image: 'assets/mbr.png'
+}
+
+const MonicaLivingRoom = {
+  name: "Monica's Living room",
+  image: 'assets/mlr.jpg'
+}
+
+const CentralPerk = {
+  name: "Central Perk",
+  image: 'assets/cp.jpg'
+}
+
+const Balcony = {
+  name: "the Balcony",
+  image: 'assets/balcony.jpg'
+}
+
+const Hall = {
+  name: "the Hall",
+  image: 'assets/stairwell.jpg'
+}
+
+const ChandlerLivingRoom = {
+  name: "Chandler's Living Room",
+  image: 'assets/clr.jpg'
+}
+
+const friend = [
+  Ross,
+  Chandler,
+  Monica,
+  Rachel,
+  Phoebe,
+  Marcel
+];
+
+const foods = [
+  Hamburger,
+  RachelTrifle,
+  Turkey,
+  RedBagel,
+  Pizza,
+  Muffin,
+  KiwiPie,
+  Sandwich,
+];
+
+const rooms = [
+  MonicaDiningRoom,
+  RossApartment,
+  ChandlerDiningRoom, 
+  RachelRoom, 
+  MonicaRoom, 
+  MonicaLivingRoom, 
+  CentralPerk, 
+  Balcony, 
+  Hall, 
+  ChandlerLivingRoom
+]
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+const mystery = {
+  friend: undefined,
+  foods: undefined,
+  rooms: undefined
```

Comment:
It's even more semantic approach to initialise mystery object with null values like this :

```
const mystery = {
  friend: null,
  foods: null,
  rooms: null
}
```

Reply:
Hmm, I understand, but is it truly necessary? I mean, yes, I could use null definitely, but why undefined is not perfect here? Sorry for the stupid question.

Reply:
It's a convention - developers have a lot of them and this is one of them. Thanks to that we can communicate without words. If I see `mystery` object with `null` values, I'm 100% sure that somewhere in your code later on you will update those values.
`undefined` on the other hand can only lead to confusions and wrong assumptions done by other developers. It's because if you see `undefined` in your code, in 99% of cases it's because function returned something in wrong way (or didn't return anything, so it returned `undefined`). At that this point we are no longer sure who've done `undefined` value - was it developer, or maybe program itself during execution? A lot of questions which we can avoid by sticking to convention of `null`. JavaScript will never return `null` by itself